src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -15,8 +22,239 @@ def __init__(self):
+        ...
+
+    def download(
+        self, file_name: Union[list[str], str], local_directory: Optional[str] = None
+    ) -> None:
+        """Download file from the server."""
+        ...
+
+
+class LocalFileTransferStrategy(FiletransferStrategy):
+    """Provides the local file transfer strategy."""
+
+    def __init__(self):
+        self.fluent_cwd = (
+            pathlib.Path(str(get_fluent_exe_path()).split("fluent")[0]) / "fluent"
+        )
+
+    def file_exists_on_remote(self, file_name: str) -> bool:
+        """Check if remote file exists.
+
+        Parameters
+        ----------
+        file_name: str
+            File name.
+
+        Returns
+        -------
+            Whether file exists.
+        """
+        full_file_name = pathlib.Path(self.fluent_cwd) / os.path.basename(file_name)
+        return full_file_name.is_file()
+
+    def upload(
+        self, file_name: Union[list[str], str], remote_file_name: Optional[str] = None
+    ) -> None:
+        local_file_name = pathlib.Path(file_name)
+        if local_file_name.exists() and local_file_name.is_file():
+            if remote_file_name:
+                shutil.copyfile(file_name, str(self.fluent_cwd / f"{remote_file_name}"))
+            else:
+                shutil.copyfile(file_name, str(self.fluent_cwd / f"{file_name}"))
+
+    def download(
+        self, file_name: Union[list[str], str], local_directory: Optional[str] = None
+    ) -> None:
+        dest_path = pathlib.Path(os.getcwd()) / f"{file_name}"
+        local_file_name = (
+            pathlib.Path(local_directory) if local_directory else dest_path
+        )
+        if local_file_name.exists() and local_file_name.samefile(file_name):
+            return
+        shutil.copyfile(file_name, str(local_file_name))
+
+
+def _get_files(file_name: Any):
+    path = "/home/runner/.local/share/ansys_fluent_core/examples"
+    if os.path.exists(path):
+        if isinstance(file_name, str):
+            files = [str(os.path.join(path, os.path.basename(file_name)))]
+        elif isinstance(
+            file_name,
+            (
+                pathlib.Path,
+                pathlib.PurePath,
+                pathlib.PosixPath,
+                pathlib.PurePosixPath,
+                pathlib.WindowsPath,
+                pathlib.PureWindowsPath,
+            ),
+        ):
+            files = [str(os.path.join(path, file_name.name))]
+        elif isinstance(file_name, list):
+            files = [
+                str(os.path.join(path, os.path.basename(file))) for file in file_name
+            ]
+    else:
+        if isinstance(file_name, str):
+            files = [file_name]
+        elif isinstance(
+            file_name,
+            (
+                pathlib.Path,
+                pathlib.PurePath,
+                pathlib.PosixPath,
+                pathlib.PurePosixPath,
+                pathlib.WindowsPath,
+                pathlib.PureWindowsPath,
+            ),
```

Comment:
```suggestion
            pathlib.PurePath
```

Reply:
We are using both `Path` and `PurePosixPath` in `fluent_container.py` and `data_transfer.py`.

So if user gives any specific type of path as follows then to handle this we added above types.

![image](https://github.com/ansys/pyfluent/assets/106588300/b2141fd3-8a01-43ab-9787-6584f188393e)

Reply:
We have added above types to fix an existing test based on `data_transfer`.

Reply:
Are they all just `PurePath`s (as per my comment above) or not? I thought that they were all in the same inheritance hierarchy.

Reply:
No, all of them are not `PurePath`, to show this I have added above screenshot.

Reply:
@seanpearsonuk You are right all of them are instance of `PurePath`. 

![image](https://github.com/ansys/pyfluent/assets/106588300/d9c6c731-8757-4729-b1d0-7a1166ee4adc)

Reply:
Done. Thank you.